sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeDirectoryClient.java

```diff
@@ -0,0 +1,420 @@
+import com.azure.storage.file.datalake.implementation.models.PathResourceType;
+import com.azure.storage.file.datalake.models.DataLakeRequestConditions;
+import com.azure.storage.file.datalake.models.PathHttpHeaders;
+import com.azure.storage.file.datalake.models.PathInfo;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Map;
+
+/**
+ * This class provides a client that contains directory operations for Azure Storage Data Lake. Operations provided by
+ * this client include creating a directory, deleting a directory, renaming a directory, setting metadata and
+ * http headers, setting and retrieving access control, getting properties and creating and deleting files and
+ * subdirectories.
+ *
+ * <p>
+ * This client is instantiated through {@link PathClientBuilder} or retrieved via
+ * {@link FileSystemClient#getDirectoryClient(String) getDirectoryClient}.
+ *
+ * <p>
+ * Please refer to the <a href=https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction?toc=%2fazure%2fstorage%2fblobs%2ftoc.json>Azure
+ * Docs</a> for more information.
+ */
+public class DataLakeDirectoryClient extends PathClient {
+    private final ClientLogger logger = new ClientLogger(DataLakeDirectoryClient.class);
+
+    private final DataLakeDirectoryAsyncClient dataLakeDirectoryAsyncClient;
+
+    DataLakeDirectoryClient(DataLakeDirectoryAsyncClient pathAsyncClient, BlockBlobClient blockBlobClient) {
+        super(pathAsyncClient, blockBlobClient);
+        this.dataLakeDirectoryAsyncClient = pathAsyncClient;
+    }
+
+    private DataLakeDirectoryClient(PathClient pathClient) {
+        super(pathClient.pathAsyncClient, pathClient.blockBlobClient);
+        this.dataLakeDirectoryAsyncClient = new DataLakeDirectoryAsyncClient(pathClient.pathAsyncClient);
+    }
+
+    /**
+     * Gets the URL of the directory represented by this client on the Data Lake service.
+     *
+     * @return the URL.
+     */
+    public String getDirectoryUrl() {
+        return getPathUrl();
+    }
+
+    /**
+     * Gets the path of this directory, not including the name of the resource itself.
+     *
+     * @return The path of the directory.
+     */
+    public String getDirectoryPath() {
+        return getObjectPath();
+    }
+
+    /**
+     * Gets the name of this directory, not including its full path.
+     *
+     * @return The name of the directory.
+     */
+    public String getDirectoryName() {
+        return getObjectName();
+    }
+
+    /**
+     * Creates a directory.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeDirectoryClient.create}
+     *
+     * <p>For more information see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create">Azure
+     * Docs</a></p>
+     *
+     * @return Information about the created directory.
+     */
+    public PathInfo create() {
```

Comment:
Same as async about moving into PathClient.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/6103